Terminal.Gui/View/Layout/ViewLayout.cs

```diff
@@ -225,6 +233,31 @@ public virtual Rect Bounds
         }
     }
 
+    /// <summary>
+    ///     The content area represent the View-relative rectangle used for this view. The area inside the view where subviews
+    ///     and content are presented. The Location is always (0,0). It will be mainly used for clipping a region.
+    /// </summary>
+    public virtual Rect ContentArea
```

Comment:
I'll study this further, but it seems to make sense to me. Using the two properties separately made it easier to manage them. Now he proposes three properties that complicate the calculations a little more but make sense.

Reply:
We really have to define these properties well. We should not confuse `ContentSize` with the `Size` of the `ScrollBar`, as they are distinct. We don't even need to use the real size of the `Size` relative to the `ScrollBar` in the `ContentSize`. That's why I propose using `ContentLocationOffset` and `ContentSizeOffset`. The first will be located from the origin of the visible area (0,0) and may be negative. The second will be the size from the origin of the `ContentLocationOffset` to the limit of the visible size. To draw we have to send the actual size of the `Bounds` that you now want to name `ContentArea`, including the `ContentLocationOffset` and the `ContentSizeOffset`. To define the `Driver.Clip` we have to send it from the origin (0,0) to the visible size. The question is whether the `ContentArea` should only contain the visible size with origin (0,0) plus the visible size or negative location plus the visible size and the ContentSizeOffset.

Reply:
Damn we only need one property which is the `ContentOffset`. This will be used to calculate the location and the size as well. I have smoke on my head. 😄